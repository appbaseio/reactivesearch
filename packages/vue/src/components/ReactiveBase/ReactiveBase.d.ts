import * as types from '../../types.ts';

export interface ReactiveBaseProps {
	app?: string;
	children?: types.children;
	credentials?: string;
	as?: string;
	headers?: types.headers;
	queryParams?: string;
	theme?: types.style;
	themePreset?: types.themePreset;
	type?: string;
	url?: string;
	mapKey?: string;
	style?: types.style;
	className?: string;
	graphQLUrl?: string;
	transformResponse?: (...args: any[]) => any;
	transformRequest?: (...args: any[]) => any;
	getSearchParams?: () => string;
	setSearchParams?: (newURL: string) => void;
	initialQueriesSyncTime?: number;
	analyticsConfig?: types.analyticsConfig;
	reactivesearchAPIConfig?: types.appbaseConfig;
	initialState?: types.children;
	analytics?: boolean;
	endpoint?: types.endpointConfig;
	contextCollector?: any;
}

declare function ReactiveBase(props: ReactiveBaseProps): JSX.Element;

export default ReactiveBase;
