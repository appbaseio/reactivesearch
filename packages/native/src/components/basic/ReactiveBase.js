import React, { Component } from 'react';
import { createProvider } from 'react-redux';
import { Container } from 'native-base';
import Appbase from 'appbase-js';

import configureStore, { storeKey } from '@appbaseio/reactivecore';

import types from '@appbaseio/reactivecore/lib/utils/types';

import ThemeProvider from '../../theme/ThemeProvider';
import theme from '../../theme';

/* use a custom store key so reactivesearch does not interfere
   with a different redux store in a nested context */
const Provider = createProvider(storeKey);

// for network debugging while development
// GLOBAL.XMLHttpRequest = GLOBAL.originalXMLHttpRequest || GLOBAL.XMLHttpRequest;

class ReactiveBase extends Component {
	constructor(props) {
		super(props);

		this.type = props.type ? props.type : '*';

		const credentials
			= props.url && props.url.trim() !== '' && !props.credentials ? null : props.credentials;

		const config = {
			url: props.url && props.url.trim() !== '' ? props.url : 'https://scalr.api.appbase.io',
			app: props.app,
			credentials,
			type: this.type,
		};

		const { headers = {} } = props;
		const appbaseRef = Appbase(config);
		this.store = configureStore({ config, appbaseRef, headers });
	}

	render() {
		return (
			<ThemeProvider theming={{ ...theme, ...this.props.theme }}>
				<Provider store={this.store}>
					<Container>{this.props.children}</Container>
				</Provider>
			</ThemeProvider>
		);
	}
}

ReactiveBase.propTypes = {
	type: types.string,
	url: types.string,
	credentials: types.string,
	headers: types.headers,
	app: types.stringRequired,
	theme: types.style,
	children: types.children,
};

export default ReactiveBase;
