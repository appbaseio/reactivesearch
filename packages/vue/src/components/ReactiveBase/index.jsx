import configureStore from '@appbaseio/reactivecore';
import VueTypes from 'vue-types';
import Appbase from 'appbase-js';
import Provider from '../Provider';
import { composeThemeObject } from '../../utils/index';
import types from '../../utils/vueTypes';
import URLParamsProvider from '../URLParamsProvider.jsx';
import getTheme from '../../styles/theme';

const URLSearchParams = require('url-search-params');

const ReactiveBase = {
	name: 'ReactiveBase',
	data() {
		this.state = {
			key: '__REACTIVE_BASE__',
		};
		return this.state;
	},
	created() {
		this.setStore(this.$props);
	},
	props: {
		app: types.stringRequired,
		analytics: VueTypes.bool.def(false),
		credentials: types.string,
		headers: types.headers,
		queryParams: types.string,
		theme: VueTypes.object.def({}),
		themePreset: VueTypes.string.def('light'),
		type: types.string,
		url: types.string,
		mapKey: types.string,
		className: types.string,
		initialState: VueTypes.object.def({}),
		transformRequest: types.func,
	},
	provide() {
		return {
			theme: composeThemeObject(getTheme(this.$props.themePreset), this.$props.theme),
			store: this.store,
		};
	},
	watch: {
		app() {
			this.updateState(this.$props);
		},
		url() {
			this.updateState(this.$props);
		},
		type() {
			this.updateState(this.$props);
		},
		credentials() {
			this.updateState(this.$props);
		},
		mapKey() {
			this.updateState(this.$props);
		},
		headers() {
			this.updateState(this.$props);
		},
	},
	methods: {
		updateState(props) {
			this.setStore(props);
			this.setState(state => ({
				key: `${state.key}-0`,
			}));
		},
		setStore(props) {
			const credentials
				= props.url && props.url.trim() !== '' && !props.credentials
					? null
					: props.credentials;
			const config = {
				url:
					props.url && props.url.trim() !== ''
						? props.url
						: 'https://scalr.api.appbase.io',
				app: props.app,
				credentials,
				type: props.type ? props.type : '*',
				transformRequest: props.transformRequest,
				analytics: props.analytics,
			};
			let queryParams = '';

			if (typeof window !== 'undefined') {
				queryParams = window.location.search;
			} else {
				queryParams = props.queryParams || '';
			}

			const params = new URLSearchParams(queryParams);
			let selectedValues = {};

			Array.from(params.keys()).forEach(key => {
				try {
					const parsedParams = JSON.parse(params.get(key));
					const selectedValue = {};
					if (parsedParams.value) {
						selectedValue.value = parsedParams.value;
					} else {
						selectedValue.value = parsedParams;
					}
					if (parsedParams.category) selectedValue.category = parsedParams.category;
					selectedValues = {
						...selectedValues,
						[key]: selectedValue,
					};
				} catch (e) {
					// Do not add to selectedValues if JSON parsing fails.
				}
			});

			const { headers = {}, themePreset } = props;
			const appbaseRef = Appbase(config);

			if (this.$props.transformRequest) {
				appbaseRef.transformRequest = this.$props.transformRequest;
			}

			const initialState = {
				config: {
					...config,
					mapKey: props.mapKey,
					themePreset,
				},
				appbaseRef,
				selectedValues,
				headers,
				...this.$props.initialState,
			};
			this.store = configureStore(initialState);
		},
	},
	render() {
		const children = this.$slots.default;
		const { headers, style, className } = this.$props;
		return (
			<Provider store={this.store}>
				<URLParamsProvider headers={headers} style={style} className={className}>
					{children}
				</URLParamsProvider>
			</Provider>
		);
	},
};
ReactiveBase.install = function(Vue) {
	Vue.component(ReactiveBase.name, ReactiveBase);
};

export default ReactiveBase;
