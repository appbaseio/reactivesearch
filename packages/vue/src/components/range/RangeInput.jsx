import VueTypes from 'vue-types';
import { helper } from '@appbaseio/reactivecore';
import { componentTypes } from '@appbaseio/reactivecore/lib/utils/constants';
import Container from '../../styles/Container';
import { connect, updateCustomQuery, isQueryIdentical } from '../../utils/index';
import ComponentWrapper from '../basic/ComponentWrapper.jsx';
import types from '../../utils/vueTypes';
import { RangeConnected as RangeSlider } from './RangeSlider.jsx';
import Input from '../../styles/Input';
import Content from '../../styles/Content';
import Flex from '../../styles/Flex';

const { checkValueChange, getClassName, getOptionsFromQuery, isEqual } = helper;

const RangeInput = {
	name: 'RangeInput',
	components: {
		RangeSlider,
	},
	inject: {
		theme: {
			from: 'theme_reactivesearch',
		},
	},
	data() {
		const state = {
			currentValue: {
				start: this.$props.range ? this.$props.range.start : 0,
				end: this.$props.range ? this.$props.range.end : 10
			},
			isStartValid: true,
			isEndValid: true
		};
		return state;
	},

	props: {
		className: {
			types: types.string,
			default: ''
		},
		defaultValue: types.range,
		validateRange: types.func,
		value: types.range,
		dataField: types.stringRequired,
		innerClass: types.style,
		range: {
			types: types.range,
			default() {
				return {
					start: 0,
					end: 10
				}
			}
		},
		rangeLabels: types.rangeLabels,
		stepValue: types.number,
		componentStyle: types.style,
		componentId: types.stringRequired,
		includeNullValues: types.bool,
		beforeValueChange: types.func,
		customQuery: types.func,
		data: types.data,
		filterLabel: types.string,
		react: types.react,
		showFilter: VueTypes.bool.def(true),
		showCheckbox: VueTypes.bool.def(true),
		title: types.title,
		URLParams: VueTypes.bool.def(false),
		sliderOptions: VueTypes.object.def({}),
		nestedField: types.string,
	},

	methods: {
		shouldUpdate(value) {
			const { validateRange } = this.$props;
			if (validateRange && value) {
				return validateRange([value.start, value.end]);
			}
			if (value) {
				return value.start <= value.end;
			}
			return false;
		},
		isControlled() {
			if (this.$props.value && this.$listeners) {
				return true;
			}
			return false;
		},
		handleChange(value, event) {
			if (this.shouldUpdate(value) && !isEqual(value, this.currentValue)) {
				switch (event) {
					case 'change':
						if(!value) {
							value = {
								start: this.$props.range ? this.$props.range.start : 0,
								end: this.$props.range ? this.$props.range.end : 10
							}
						}
						this.$data.currentValue = {...value};
						this.$emit('change', this.$data.currentValue);
						break;
					case 'value-change':
						this.$emit('valueChange', this.$data.currentValue);
						this.$emit('value-change', this.$data.currentValue);
						break;
					default:
						this.$data.currentValue = {...value};
						break;
				}
			}
		},
		handleOnChange(value) {
			this.handleChange(value, 'change');
		},
		handleValueChange(value) {
			this.handleChange(value, 'value-change');
		},
		handleInputChange(e) {
			const { name, value } = e.target;
			if (Number.isNaN(value)) {
				if (name === 'start') {
					this.$data.isStartValid = false;
				} else {
					this.$data.isEndValid = false;
				}
			} else {
				if (name === 'start' && !this.$data.isStartValid) {
					this.$data.isStartValid = true;
				} else if (name === 'end' && !this.$data.isEndValid) {
					this.$data.isEndValid = true;
				}
			}

			if (this.$data.isStartValid && this.$data.isEndValid) {
				if (name === 'start') {
					this.handleChange({
						start: Number(value),
						end: this.$data.currentValue.end
					}, 'change');
				} else {
					this.handleChange({
						start: this.$data.currentValue.start,
						end: Number(value)
					}, 'change');
				}
			}
		}
	},
	watch: {
		defaultValue(newVal, oldVal) {
			if (oldVal.start !== newVal.start || oldVal.end !== newVal.end) {
				this.handleChange(newVal)
			}
		},
		value(newVal, oldVal) {
			if (!isEqual(newVal, oldVal)) {
				if (this.isControlled()) {
					this.handleChange(newVal, 'change');
				}
			}
		},
	},
	created() {
		if (this.$props.defaultValue && this.$props.defaultValue.start && this.$props.defaultValue.end) {
			this.handleChange(this.$props.defaultValue)
		}
	},
	render() {
		const {
			className,
			dataField,
			range,
			rangeLabels,
			componentId,
			innerClass,
			stepValue,
			componentStyle,
			themePreset,
			includeNullValues,
			beforeValueChange,
			customQuery,
			data,
			filterLabel,
			react,
			showFilter,
			showCheckbox,
			title,
			URLParams,
			sliderOptions,
			nestedField,
		} = this.$props;
		return (
			<Container style={componentStyle} class={className}>
				<RangeSlider
					componentId={componentId}
					value={{
						start: this.currentValue.start,
						end: this.currentValue.end
					}}
					range={range}
					dataField={dataField}
					rangeLabels={rangeLabels}
					includeNullValues={includeNullValues}
					beforeValueChange={beforeValueChange}
					customQuery={customQuery}
					data={data}
					filterLabel={filterLabel}
					react={react}
					showFilter={showFilter}
					showCheckbox={showCheckbox}
					title={title}
					URLParams={URLParams}
					sliderOptions={sliderOptions}
					nestedField={nestedField}
					on-change={this.handleOnChange}
					on-value-change={this.handleValueChange}
				/>
				<Flex class={getClassName(innerClass, 'input-container') || ''}>
					<Flex direction="column" flex={2}>
						<Input
							key={`${componentId}-start-value`}
							name="start"
							type="number"
							onChange={this.handleInputChange}
							step={stepValue}
							themePreset={themePreset}
							aria-label={`${componentId}-start-input`}
							min={this.$props.range ? this.$props.range.start : 0}
							class={getClassName(innerClass, 'input') || ''}
							alert={!this.isStartValid}
							{
								...{
									domProps: {
										value: this.currentValue.start
									}
								}
							}
						/>
						{!this.isStartValid && (
							<Content alert>Input range is invalid</Content>
						)}
					</Flex>
					<Flex justifyContent="center" alignItems="center" flex={1}>
						-
					</Flex>
					<Flex direction="column" flex={2}>
						<Input
							key={`${componentId}-end-value`}
							name="end"
							type="number"
							onChange={this.handleInputChange}
							step={stepValue}
							themePreset={themePreset}
							aria-label={`${componentId}-end-input`}
							max={this.$props.range ? this.$props.range.end : 10}
							class={getClassName(innerClass, 'input') || ''}
							alert={!this.isEndValid}
							{
								...{
									domProps: {
										value: this.currentValue.end
									}
								}
							}
						/>
						{!this.isEndValid && <Content alert>Input range is invalid</Content>}
					</Flex>
				</Flex>
			</Container>
		);
	},
};

const mapStateToProps = (state, props) => ({
	themePreset: state.config.themePreset,
});

const RangeConnected = ComponentWrapper(connect(mapStateToProps, {})(RangeInput), {
	componentType: componentTypes.rangeInput,
});

RangeInput.install = function(Vue) {
	Vue.component(RangeInput.name, RangeConnected);
};

// Add componentType for SSR
RangeInput.componentType = componentTypes.rangeInput;

export default RangeInput;
