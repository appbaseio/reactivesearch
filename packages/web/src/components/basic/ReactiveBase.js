/* eslint-disable global-require */

import React, { Component } from 'react';
import { createProvider } from 'react-redux';
import Appbase from 'appbase-js';
import { ThemeProvider } from 'emotion-theming';

import configureStore, { storeKey } from '@appbaseio/reactivecore';
import { checkSomePropChange } from '@appbaseio/reactivecore/lib/utils/helper';
import types from '@appbaseio/reactivecore/lib/utils/types';
import URLParamsProvider from './URLParamsProvider';

import getTheme from '../../styles/theme';
import { composeThemeObject } from '../../utils';

const URLSearchParams = require('url-search-params');

/* use a custom store key so reactivesearch does not interfere
   with a different redux store in a nested context */
const Provider = createProvider(storeKey);

class ReactiveBase extends Component {
	constructor(props) {
		super(props);

		this.state = {
			key: '__REACTIVE_BASE__',
		};

		this.setStore(props);
	}

	componentWillReceiveProps(nextProps) {
		checkSomePropChange(
			this.props,
			nextProps,
			['app', 'url', 'type', 'credentials', 'mapKey', 'headers'],
			() => {
				this.setStore(nextProps);
				this.setState((state => ({
					key: `${state.key}-0`,
				})));
			},
		);
	}

	componentDidCatch() {
		console.error(
			'An error has occured. You\'re using Reactivesearch Version:',
			`${process.env.VERSION || require('../../../package.json').version}.`,
			'If you think this is a problem with Reactivesearch, please try updating',
			'to the latest version. If you\'re already at the latest version, please open',
			'an issue at https://github.com/appbaseio/reactivesearch/issues',
		);
	}

	setStore = (props) => {
		this.type = props.type ? props.type : '*';

		const credentials = props.url && props.url.trim() !== '' && !props.credentials
			? null
			: props.credentials;

		const config = {
			url: props.url && props.url.trim() !== '' ? props.url : 'https://scalr.api.appbase.io',
			app: props.app,
			credentials,
			type: this.type,
			transformRequest: props.transformRequest,
			analytics: props.analytics,
		};

		let queryParams = '';
		if (typeof window !== 'undefined') {
			queryParams = window.location.search;
		} else {
			queryParams = props.queryParams || '';
		}

		const params = new URLSearchParams(queryParams);
		let selectedValues = {};

		try {
			Array.from(params.keys()).forEach((key) => {
				selectedValues = {
					...selectedValues,
					[key]: { value: JSON.parse(params.get(key)) },
				};
			});
		} catch (e) {
			console.error('REACTIVESEARCH - An error occured while parsing the URL state.', e);
			selectedValues = {};
		}

		const { headers = {}, themePreset } = props;
		const appbaseRef = Appbase(config);

		const initialState = {
			config: { ...config, mapKey: props.mapKey, themePreset },
			appbaseRef,
			selectedValues,
			headers,
			...this.props.initialState,
		};
		this.store = configureStore(initialState);
	};

	render() {
		const theme = composeThemeObject(
			getTheme(this.props.themePreset),
			this.props.theme,
		);

		return (
			<ThemeProvider
				theme={theme}
				key={this.state.key}
			>
				<Provider store={this.store}>
					<URLParamsProvider
						headers={this.props.headers}
						style={this.props.style}
						className={this.props.className}
					>
						{this.props.children}
					</URLParamsProvider>
				</Provider>
			</ThemeProvider>
		);
	}
}

ReactiveBase.defaultProps = {
	theme: {},
	themePreset: 'light',
	initialState: {},
	analytics: false,
};

ReactiveBase.propTypes = {
	app: types.stringRequired,
	children: types.children,
	credentials: types.string,
	headers: types.headers,
	queryParams: types.string,
	theme: types.style,
	themePreset: types.themePreset,
	type: types.string,
	url: types.string,
	transformRequest: types.func,
	mapKey: types.string,
	style: types.style,
	className: types.string,
	initialState: types.children,
	analytics: types.bool,
};

export default ReactiveBase;
