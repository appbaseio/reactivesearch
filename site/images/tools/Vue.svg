<?xml version="1.0" encoding="UTF-8"?>
<svg width="142px" height="117px" viewBox="0 0 142 117" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <!-- Generator: sketchtool 52.4 (67378) - http://www.bohemiancoding.com/sketch -->
    <title>F5968638-E0A9-437B-8498-2A31D8EDA594@3x</title>
    <desc>Created with sketchtool.</desc>
    <g id="Page-1" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
        <g id="Reactive-Search-//-Tools" transform="translate(-801.000000, -485.000000)" fill-rule="nonzero" stroke="#FFFFFF">
            <g id="Group-5" transform="translate(784.000000, 488.000000)">
                <g id="Group-13" transform="translate(20.000000, 0.000000)">
                    <polygon id="Path-4" stroke-width="3" opacity="0.46765718" points="4.40536496e-13 -1.42108547e-14 52.5512501 -1.42108547e-14 67.7970088 25.5378091 83.1987012 -1.42108547e-14 135.434835 -1.42108547e-14 67.7174177 111"></polygon>
                    <path d="M28,-1.49213975e-13 L52.2655434,-1.49213975e-13 L67.2351697,25.5378091 L83.4652594,-1.49213975e-13 C99.2751946,-1.49213975e-13 107.180162,-1.49213975e-13 107.180162,-1.49213975e-13 C107.180162,-1.49213975e-13 93.9834685,22.8776207 67.5900811,68.632862 L28,-1.49213975e-13 Z" id="Path-15" stroke-width="5"></path>
                </g>
            </g>
        </g>
    </g>
</svg>