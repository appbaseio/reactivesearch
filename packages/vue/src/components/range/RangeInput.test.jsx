import { mount } from '@vue/test-utils';
import Vue from 'vue';
import { RangeConnected as RangeInput } from './RangeInput.jsx';
import ReactiveBase from '../ReactiveBase/index.jsx';

Vue.component('vue-slider-component', require('vue-slider-component'));

describe('RangeInput', () => {
	it('should render range with slider', () => {
		const wrapper = mount({
			name: 'TestComponent',
			render() {
				return (
					<ReactiveBase app="test" url="https://foo:bar@localhost:800">
						<RangeInput
							mode="test"
							componentId="authors"
							dataField="ratings_count"
							range={{
								start: 30000,
								end: 50000,
							}}
						/>
					</ReactiveBase>
				);
			},
		});

		expect(wrapper.element).toMatchSnapshot();
	});

	it('should render title', () => {
		const wrapper = mount({
			name: 'TestComponent',
			render() {
				return (
					<ReactiveBase app="test" url="https://foo:bar@localhost:800">
						<RangeInput
							mode="test"
							componentId="authors"
							dataField="ratings_count"
							range={{
								start: 30000,
								end: 50000,
							}}
							title="Test Title"
						/>
					</ReactiveBase>
				);
			},
		});

		expect(wrapper.element).toMatchSnapshot();
	});

	it('should render range labels', () => {
		const wrapper = mount({
			name: 'TestComponent',
			render() {
				return (
					<ReactiveBase app="test" url="https://foo:bar@localhost:800">
						<RangeInput
							mode="test"
							componentId="authors"
							dataField="ratings_count"
							range={{
								start: 20000,
								end: 50000,
							}}
							rangeLabels={{
								start: '2,000',
								end: '50,000',
							}}
						/>
					</ReactiveBase>
				);
			},
		});

		expect(wrapper.element).toMatchSnapshot();
	});

	it('should select default value', () => {
		const wrapper = mount({
			name: 'TestComponent',
			render() {
				return (
					<ReactiveBase app="test" url="https://foo:bar@localhost:800">
						<RangeInput
							mode="test"
							componentId="authors"
							dataField="ratings_count"
							range={{
								start: 3000,
								end: 5500,
							}}
							rangeLabels={{
								start: '3K',
								end: '5.5K',
							}}
							defaultValue={{
								start: 3200,
								end: 4500,
							}}
						/>
					</ReactiveBase>
				);
			},
		});

		expect(wrapper.element).toMatchSnapshot();
	});
});
