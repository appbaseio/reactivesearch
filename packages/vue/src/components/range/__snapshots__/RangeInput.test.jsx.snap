// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RangeInput should render range labels 1`] = `
<div
  class="css-nw180v"
>
  <div
    class="css-m1gst5"
    mode="test"
  >
    <div
      class="css-m1gst5"
      mode="test"
    >
      <div
        class="css-1sjm8x1"
      >
        <div
          class="vue-slider vue-slider-ltr"
          style="padding: 10px 0px; width: auto; height: 4px;"
        >
          <div
            class="vue-slider-rail"
          >
            <div
              class="vue-slider-process"
              style="height: 100%; top: 0px; left: 0%; width: 100%; transition-property: width,left; transition-duration: 0.5s;"
            />
            <div
              aria-orientation="horizontal"
              aria-valuemax="50000"
              aria-valuemin="20000"
              aria-valuenow="20000"
              aria-valuetext="20000"
              class="vue-slider-dot"
              role="slider"
              style="width: 20px; height: 20px; transform: translate(-50%, -50%); top: 50%; left: 0%; transition: left 0.5s;"
              tabindex="0"
            >
              <div
                class="vue-slider-dot-handle"
              />
              <div
                class="vue-slider-dot-tooltip vue-slider-dot-tooltip-top vue-slider-dot-tooltip-show"
              >
                <div
                  class="vue-slider-dot-tooltip-inner vue-slider-dot-tooltip-inner-top"
                >
                  <span
                    class="vue-slider-dot-tooltip-text"
                  >
                    20000
                  </span>
                </div>
              </div>
            </div>
            <div
              aria-orientation="horizontal"
              aria-valuemax="50000"
              aria-valuemin="20000"
              aria-valuenow="50000"
              aria-valuetext="50000"
              class="vue-slider-dot"
              role="slider"
              style="width: 20px; height: 20px; transform: translate(-50%, -50%); top: 50%; left: 100%; transition: left 0.5s;"
              tabindex="0"
            >
              <div
                class="vue-slider-dot-handle"
              />
              <div
                class="vue-slider-dot-tooltip vue-slider-dot-tooltip-top vue-slider-dot-tooltip-show"
              >
                <div
                  class="vue-slider-dot-tooltip-inner vue-slider-dot-tooltip-inner-top"
                >
                  <span
                    class="vue-slider-dot-tooltip-text"
                  >
                    50000
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="label-container"
        >
          <label
            class="range-label-left"
          >
            2,000
          </label>
          <label
            class="range-label-right"
          >
            50,000
          </label>
        </div>
      </div>
    </div>
    <div
      class="css-kgsg89"
    >
      <div
        class="css-d2gq24"
        direction="column"
        flex="2"
      >
        <input
          aria-label="authors-start-input"
          class="css-1n1dm2c"
          min="20000"
          name="start"
          type="number"
          value="20000"
        />
      </div>
      <div
        alignitems="center"
        class="css-6c29fo"
        flex="1"
        justifycontent="center"
      >
        -
      </div>
      <div
        class="css-d2gq24"
        direction="column"
        flex="2"
      >
        <input
          aria-label="authors-end-input"
          class="css-1n1dm2c"
          max="50000"
          name="end"
          type="number"
          value="50000"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`RangeInput should render range with slider 1`] = `
<div
  class="css-nw180v"
>
  <div
    class="css-m1gst5"
    mode="test"
  >
    <div
      class="css-m1gst5"
      mode="test"
    >
      <div
        class="css-1sjm8x1"
      >
        <div
          class="vue-slider vue-slider-ltr"
          style="padding: 10px 0px; width: auto; height: 4px;"
        >
          <div
            class="vue-slider-rail"
          >
            <div
              class="vue-slider-process"
              style="height: 100%; top: 0px; left: 0%; width: 100%; transition-property: width,left; transition-duration: 0.5s;"
            />
            <div
              aria-orientation="horizontal"
              aria-valuemax="50000"
              aria-valuemin="30000"
              aria-valuenow="30000"
              aria-valuetext="30000"
              class="vue-slider-dot"
              role="slider"
              style="width: 20px; height: 20px; transform: translate(-50%, -50%); top: 50%; left: 0%; transition: left 0.5s;"
              tabindex="0"
            >
              <div
                class="vue-slider-dot-handle"
              />
              <div
                class="vue-slider-dot-tooltip vue-slider-dot-tooltip-top vue-slider-dot-tooltip-show"
              >
                <div
                  class="vue-slider-dot-tooltip-inner vue-slider-dot-tooltip-inner-top"
                >
                  <span
                    class="vue-slider-dot-tooltip-text"
                  >
                    30000
                  </span>
                </div>
              </div>
            </div>
            <div
              aria-orientation="horizontal"
              aria-valuemax="50000"
              aria-valuemin="30000"
              aria-valuenow="50000"
              aria-valuetext="50000"
              class="vue-slider-dot"
              role="slider"
              style="width: 20px; height: 20px; transform: translate(-50%, -50%); top: 50%; left: 100%; transition: left 0.5s;"
              tabindex="0"
            >
              <div
                class="vue-slider-dot-handle"
              />
              <div
                class="vue-slider-dot-tooltip vue-slider-dot-tooltip-top vue-slider-dot-tooltip-show"
              >
                <div
                  class="vue-slider-dot-tooltip-inner vue-slider-dot-tooltip-inner-top"
                >
                  <span
                    class="vue-slider-dot-tooltip-text"
                  >
                    50000
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-kgsg89"
    >
      <div
        class="css-d2gq24"
        direction="column"
        flex="2"
      >
        <input
          aria-label="authors-start-input"
          class="css-1n1dm2c"
          min="30000"
          name="start"
          type="number"
          value="30000"
        />
      </div>
      <div
        alignitems="center"
        class="css-6c29fo"
        flex="1"
        justifycontent="center"
      >
        -
      </div>
      <div
        class="css-d2gq24"
        direction="column"
        flex="2"
      >
        <input
          aria-label="authors-end-input"
          class="css-1n1dm2c"
          max="50000"
          name="end"
          type="number"
          value="50000"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`RangeInput should render title 1`] = `
<div
  class="css-nw180v"
>
  <div
    class="css-m1gst5"
    mode="test"
  >
    <div
      class="css-m1gst5"
      mode="test"
    >
      <h2
        class="css-dysiaa"
      >
        Test Title
      </h2>
      <div
        class="css-1sjm8x1"
      >
        <div
          class="vue-slider vue-slider-ltr"
          style="padding: 10px 0px; width: auto; height: 4px;"
        >
          <div
            class="vue-slider-rail"
          >
            <div
              class="vue-slider-process"
              style="height: 100%; top: 0px; left: 0%; width: 100%; transition-property: width,left; transition-duration: 0.5s;"
            />
            <div
              aria-orientation="horizontal"
              aria-valuemax="50000"
              aria-valuemin="30000"
              aria-valuenow="30000"
              aria-valuetext="30000"
              class="vue-slider-dot"
              role="slider"
              style="width: 20px; height: 20px; transform: translate(-50%, -50%); top: 50%; left: 0%; transition: left 0.5s;"
              tabindex="0"
            >
              <div
                class="vue-slider-dot-handle"
              />
              <div
                class="vue-slider-dot-tooltip vue-slider-dot-tooltip-top vue-slider-dot-tooltip-show"
              >
                <div
                  class="vue-slider-dot-tooltip-inner vue-slider-dot-tooltip-inner-top"
                >
                  <span
                    class="vue-slider-dot-tooltip-text"
                  >
                    30000
                  </span>
                </div>
              </div>
            </div>
            <div
              aria-orientation="horizontal"
              aria-valuemax="50000"
              aria-valuemin="30000"
              aria-valuenow="50000"
              aria-valuetext="50000"
              class="vue-slider-dot"
              role="slider"
              style="width: 20px; height: 20px; transform: translate(-50%, -50%); top: 50%; left: 100%; transition: left 0.5s;"
              tabindex="0"
            >
              <div
                class="vue-slider-dot-handle"
              />
              <div
                class="vue-slider-dot-tooltip vue-slider-dot-tooltip-top vue-slider-dot-tooltip-show"
              >
                <div
                  class="vue-slider-dot-tooltip-inner vue-slider-dot-tooltip-inner-top"
                >
                  <span
                    class="vue-slider-dot-tooltip-text"
                  >
                    50000
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-kgsg89"
    >
      <div
        class="css-d2gq24"
        direction="column"
        flex="2"
      >
        <input
          aria-label="authors-start-input"
          class="css-1n1dm2c"
          min="30000"
          name="start"
          type="number"
          value="30000"
        />
      </div>
      <div
        alignitems="center"
        class="css-6c29fo"
        flex="1"
        justifycontent="center"
      >
        -
      </div>
      <div
        class="css-d2gq24"
        direction="column"
        flex="2"
      >
        <input
          aria-label="authors-end-input"
          class="css-1n1dm2c"
          max="50000"
          name="end"
          type="number"
          value="50000"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`RangeInput should select default value 1`] = `
<div
  class="css-nw180v"
>
  <div
    class="css-m1gst5"
    mode="test"
  >
    <div
      class="css-m1gst5"
      mode="test"
    >
      <div
        class="css-1sjm8x1"
      >
        <div
          class="vue-slider vue-slider-ltr"
          style="padding: 10px 0px; width: auto; height: 4px;"
        >
          <div
            class="vue-slider-rail"
          >
            <div
              class="vue-slider-process"
              style="height: 100%; top: 0px; left: 8%; width: 52%; transition-property: width,left; transition-duration: 0.5s;"
            />
            <div
              aria-orientation="horizontal"
              aria-valuemax="5500"
              aria-valuemin="3000"
              aria-valuenow="3200"
              aria-valuetext="3200"
              class="vue-slider-dot"
              role="slider"
              style="width: 20px; height: 20px; transform: translate(-50%, -50%); top: 50%; left: 8%; transition: left 0.5s;"
              tabindex="0"
            >
              <div
                class="vue-slider-dot-handle"
              />
              <div
                class="vue-slider-dot-tooltip vue-slider-dot-tooltip-top vue-slider-dot-tooltip-show"
              >
                <div
                  class="vue-slider-dot-tooltip-inner vue-slider-dot-tooltip-inner-top"
                >
                  <span
                    class="vue-slider-dot-tooltip-text"
                  >
                    3200
                  </span>
                </div>
              </div>
            </div>
            <div
              aria-orientation="horizontal"
              aria-valuemax="5500"
              aria-valuemin="3000"
              aria-valuenow="4500"
              aria-valuetext="4500"
              class="vue-slider-dot"
              role="slider"
              style="width: 20px; height: 20px; transform: translate(-50%, -50%); top: 50%; left: 60%; transition: left 0.5s;"
              tabindex="0"
            >
              <div
                class="vue-slider-dot-handle"
              />
              <div
                class="vue-slider-dot-tooltip vue-slider-dot-tooltip-top vue-slider-dot-tooltip-show"
              >
                <div
                  class="vue-slider-dot-tooltip-inner vue-slider-dot-tooltip-inner-top"
                >
                  <span
                    class="vue-slider-dot-tooltip-text"
                  >
                    4500
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="label-container"
        >
          <label
            class="range-label-left"
          >
            3K
          </label>
          <label
            class="range-label-right"
          >
            5.5K
          </label>
        </div>
      </div>
    </div>
    <div
      class="css-kgsg89"
    >
      <div
        class="css-d2gq24"
        direction="column"
        flex="2"
      >
        <input
          aria-label="authors-start-input"
          class="css-1n1dm2c"
          min="3000"
          name="start"
          type="number"
          value="3200"
        />
      </div>
      <div
        alignitems="center"
        class="css-6c29fo"
        flex="1"
        justifycontent="center"
      >
        -
      </div>
      <div
        class="css-d2gq24"
        direction="column"
        flex="2"
      >
        <input
          aria-label="authors-end-input"
          class="css-1n1dm2c"
          max="5500"
          name="end"
          type="number"
          value="4500"
        />
      </div>
    </div>
  </div>
</div>
`;
